### PR TITLE
I1/event agggregates alternative solution

### PR DESCRIPTION
**Pull request by polomani**

### CODE REVIEW

File: src/resources/events/eventsApi.ts [hunk 1]
```diff
@@ -53,12 +55,22 @@ export class EventsAPI extends BaseResourceAPI<CogniteEvent> {
...
-  ): Promise<EventAggregate[]> => {
-    return super.aggregateEndpoint(query);
+  public aggregate = <
+    T extends EventAggregateQuery | EventAggregateUniqueValues
+  >(
+    query: T
+  ) => {
+    return super.aggregateEndpoint<
+      T,
+      T extends EventAggregateUniqueValues
```
Comment (BugGambit) [inline src/resources/events/eventsApi.ts:70]: How will we easliy expand this for new aggregation methods?

File: src/__tests__/resources/events.integration.spec.ts [hunk 2]
```diff
@@ -95,6 +95,19 @@ describe('Events integration test', () => {
     expect(aggregates[0].count).toBeDefined();
   });
 
+  test('values aggregate', async () => {
+    const aggregates = await client.events.aggregate({
+      filter: {
+        source: 'WORKMATE',
+      },
+      fields: ['type'],
+      aggregate: 'values',
```
Comment (hhenriques) [inline src/__tests__/resources/events.integration.spec.ts:104]: Not a fan of the usability here. The user has to know what fields are compatible with each aggregate type.

Comment (polomani) [reply src/__tests__/resources/events.integration.spec.ts:104]: but it's typed, you cannot randomly choose a string

**Comment by polomani** [comment]
we'll go for methods approach

**Pull request closed without merging**